packages/next-swc/crates/next-api/src/app.rs

```diff
@@ -894,8 +899,26 @@ impl AppEndpoint {
             ));
             server_assets.push(app_build_manifest_output);
 
+            // polyfill-nomodule.js is a pre-compiled asset distributed as part of next,
+            // load it as a RawModule.
+            let next_package = get_next_package(this.app_project.project().project_path());
+            let polyfill_source_path =
+                next_package.join("dist/build/polyfills/polyfill-nomodule.js".to_string());
+            let polyfill_module = RawModule::new(Vc::upcast(FileSource::new(polyfill_source_path)));
+            let polyfill_output_path =
+                client_chunking_context.chunk_path(polyfill_module.ident(), ".js".to_string());
+            let polyfill_output_asset =
+                VirtualOutputAsset::new(polyfill_output_path, polyfill_module.content());
```

Comment:
It doesn't really matter here, but you could change it to:

```suggestion
            let polyfill_output_asset =
                RawOutput::new(polyfill_output_path, polyfill_module);
```

If you give `RawOutput` an additional `path` argument, which it reports as `ident()` `AssetIdent::from_path(path)`. That would allow the `RawOutput` to "serve" a `Source` under a specific path.

---

The motivation behind that:

Eventually we want to have lazy compilation. That would mean instead of emitting all OutputAssets directly to disk we would directly serve them on request. That would mean `Asset::content()` on OutputAssets is only called when the file is actually requested. So we want to delay as much work as possible to this point. e. g. Chunks only generate the content in the `content()` method, so code generation is skipped until this point.
By having `RawOutput::new(polyfill_output_path, polyfill_module)` the `polyfill_module.content()` method would only be called when the file is requested. So that would cause the polyfill file not to be read until it's actually requested from the browser.

Reply:
Because this requires changes to `turbopack_core`, I'm going to leave it as `VirtualOutputAsset` in this PR, but I'll make a separate PR for the turbo repo to add a path to `RawOutput`, and a follow-up to switch this to `RawOutput`.